core/src/HIP/Kokkos_HIP_ZeroMemset.hpp

```diff
@@ -32,8 +32,7 @@ struct ZeroMemset<HIP> {
     // in ROCm <= 6.2.0, hipMemsetAsync on a host-allocated pointer
     // returns an invalid value error, but accessing the data via a
     // GPU kernel works.
-#if defined(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY) && \
-    defined(KOKKOS_ARCH_AMD_GFX942)
+#if defined(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY)
```

Comment:
Isn't `KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY` being removed as part of this PR?
Should this be ifdef'ed on `KOKKOS_ARCH_AMD_GFX942_APU` instead?

Reply:
No I want that as an internal typedef for all APU versions. I.e. if there ever is another AMD APU we would use the same option, even though the arch is different. I just don't really want it as a CMake option.